### PR TITLE
feat: add get wallet for auth data

### PR DESCRIPTION
**Pull request by infantesimone**
#### Motivation and Context

The goal of thi PR is to add API dedicated to eCommerce to retrive wallet data useful to handle authorization request with eCommerce, so `contractId`, `bin` and `brand`.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/services/WalletService.kt [hunk 1]
```diff
@@ -363,6 +363,13 @@ class WalletService(
         return walletRepository.findByUserId(userId.toString()).collectList().map { toWallets(it) }
     }
 
+    fun findWalletAuthData(walletId: UUID): Mono<WalletAuthDataDto> {
```
Comment (Fernando-Granato) [inline src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:366]: You can use as an input parameter in the service the @ValueObject WalletId

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt [hunk 2]
```diff
@@ -239,6 +239,25 @@ class WalletControllerTest {
             .json(jsonToTest)
     }
 
+    @Test
+    fun testGetWalletAuthDataSuccess() = runTest {
```
Comment (Fernando-Granato) [inline src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt:243]: You can add a test case on the WalletNotFoundException

**Review by Fernando-Granato — COMMENTED** [review]
Just a few small comments

File: src/main/kotlin/it/pagopa/wallet/services/WalletService.kt (hunk 1 above)
Comment (infantesimone) [reply src/main/kotlin/it/pagopa/wallet/services/WalletService.kt]: done :-)

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt (hunk 2 above)
Comment (infantesimone) [reply src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt:244]: correct :-) done

**Comment by infantesimone** [comment]
the coverage is not great, it depends on not having tested the cases with a details other than creditCard. I would defer these tests until we introduce the other methods

**Review by Fernando-Granato — APPROVED** [review]

**Review by ciuffagianluca — APPROVED** [review]

**Pull request merged**